fdbclient/DatabaseConfiguration.cpp

```diff
@@ -632,6 +632,57 @@ std::set<AddressExclusion> DatabaseConfiguration::getExcludedServers() const {
 	return addrs;
 }
 
+// checks if the locality is excluded or not by checking if the key is present.
+bool DatabaseConfiguration::isExcludedLocality(const LocalityData& locality) const {
+	std::map<std::string, std::string> localityData = locality.getAllData();
+	for (const auto& l : localityData) {
+		if (get(StringRef(encodeExcludedLocalityKey(ExcludeLocalityPrefix.toString() + l.first + ":" + l.second)))
+		        .present() ||
+		    get(StringRef(encodeFailedLocalityKey(ExcludeLocalityPrefix.toString() + l.first + ":" + l.second)))
+		        .present()) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+// checks if this machineid of given locality is excluded.
+// A machine can be excluded either as part of dcid exclustion or zoneid exclusion
```

Comment:
Should we update the comments here as the below function only checked for machine_id but not dc_id and zone_id?